task/thermal/src/control.rs

```diff
@@ -345,6 +359,65 @@ impl Max31790State {
     }
 }
 
+/// Tracks whether a EMC2305 fan controller has been initialized, and
+/// initializes it on demand when accessed, if necessary.
+///
+/// This is copy-pasted from [`Max31790`]
+pub(crate) struct Emc2305State {
+    emc2305: Emc2305,
+    fan_count: u8,
+    initialized: bool,
+}
+
+impl Emc2305State {
+    #[allow(dead_code)]
+    pub(crate) fn new(dev: &I2cDevice, fan_count: u8) -> Self {
+        let mut this = Self {
+            emc2305: Emc2305::new(dev),
+            fan_count,
+            initialized: false,
+        };
+        // When we first start up, try to initialize the fan controller a few
+        // times, in case there's a transient I2C error.
+        for remaining in (0..3).rev() {
+            if this.initialize().is_ok() {
+                break;
+            }
+            ringbuf_entry!(Trace::FanControllerInitRetry { remaining });
+        }
```

Comment:
take it or leave it: maybe instead of just copying this from the `Max31790` implementation, maybe we should just have a 
```rust
fn retry_init(init: impl FnMut() -> Result<...>) {
    for remaining in (0..3).rev() {
        if init().is_ok() {
            break;
        }
        ringbuf_entry!(Trace::FanControllerInitRetry { remaining });
    }
}
```
that they can both use?

Reply:
Sure!